route_following_plugin/include/route_following_plugin.h

```diff
@@ -121,20 +103,31 @@ namespace route_following_plugin
          */
         bool identifyLaneChange(lanelet::routing::LaneletRelations relations, int target_id);
 
+        void set_maneuver_start_dist(cav_msgs::Maneuver& maneuver, double start_dist);
+
         /**
          * \brief Service callback for arbitrator maneuver planning
          * \param req Plan maneuver request
          * \param resp Plan maneuver response with a list of maneuver plan
          * \return If service call successed
          */
         bool plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp);
+
+        double get_final_speed(cav_msgs::Maneuver maneuver);
+
+        double get_end_dist(cav_msgs::Maneuver maneuver);
+
         /**
          * \brief Given a Lanelet, find it's associated Speed Limit from the vector map
          * \param llt Constant Lanelet object
          * \return value of speed limit as a double, returns default value of 25 mph
          */
         double findSpeedLimit(const lanelet::ConstLanelet& llt);
 
+        void route_cb();
```

Comment:
comments needed. Also it seems like there is a mixing of camelCase and snake_case method names in this file which is a purely stylistic issue but it improves readability if everything in the same file follows similar formatting.